src/main/java/com/puppycrawl/tools/checkstyle/checks/design/MutableExceptionCheck.java

```diff
@@ -66,6 +66,132 @@
  * <pre>
  * &lt;module name=&quot;MutableException&quot;/&gt;
  * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * class FirstClass extends Exception {
+ *   private int code; // OK, class name doesn't match with default pattern
+ *
+ *   public FirstClass() {
+ *     code = 1;
+ *   }
+ * }
+ *
+ * class MyException extends Exception {
+ *   private int code; // violation, The field 'code' must be declared final
+ *
+ *   public MyException() {
+ *     code = 2;
+ *   }
+ * }
+ *
+ * class MyThrowable extends Throwable {
+ *    final int code; // OK
+ *    String message; // violation, The field 'message' must be declared final
+ *
+ *    public MyThrowable(int code, String message) {
+ *      this.code = code;
+ *      this.message = message;
+ *    }
+ * }
+ *
+ * class BadException extends java.lang.Exception {
+ *   int code; // violation, The field 'code' must be declared final
+ *
+ *   public BadException(int code) {
+ *     this.code = code;
+ *   }
+ * }
+ * </pre>
+ * <p>
+ * To configure the check so that it only checks for class names
+ * that match pattern:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;MutableException&quot;&gt;
+ *   &lt;property name=&quot;format&quot; value=&quot;^.*Exception$&quot;/&gt;
+ * &lt;/module&gt;
+ * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * class FirstClass extends Exception {
+ *   private int code; // OK, class name doesn't match with given pattern
+ *
+ *   public FirstClass() {
+ *     code = 1;
+ *   }
+ * }
+ *
+ * class MyException extends Exception {
+ *   private int code; // violation, The field 'code' must be declared final
+ *
+ *   public MyException() {
+ *     code = 2;
+ *   }
+ * }
+ *
+ * class MyThrowable extends Throwable {
+ *   final int code; // OK, class name doesn't match with given pattern
+ *   String message; // OK, class name doesn't match with given pattern
```

Comment:
Please share output of CLI.
This should be violation, we extends property will catch it.

Reply:
I have added all the outputs in PR description

Reply:
It will not catch as i haven't specify that property in my config.

Reply:
Default is still in play, please share CLI out put

Reply:
Please share output in text form to let me quickly reuse your execution and confirm that it is correct.
Share it in format as suggested at https://checkstyle.org/report_issue.html#How_to_report_a_bug.3F

Reply:
@romani, here it is

PS D:\Arin\PlayGround> type config.xml     
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <module name="TreeWalker">
    <module name="MutableException">
        <property name="format" value="^.*Exception$" />
    </module>
  </module>
</module>
PS D:\Arin\PlayGround> type FirstClass.java
class FirstClass extends Exception {
  private int code;

  public FirstClass() {
    code = 1;
  }
}

class MyException extends Exception {
  private int code;

  public MyException() {
    code = 2;
  }
}

class MyThrowable extends Throwable {
  final int code;
  String message;

  public MyThrowable(int code, String message) {
    this.code = code;
    this.message = message;
  }
}

class BadException extends java.lang.Exception {
  int code;

  public BadException(int code) {
    this.code = code;
  }
}
PS D:\Arin\PlayGround> java -jar checkstyle-10.5-SNAPSHOT-all.jar -c config.xml FirstClass.java
Starting audit...
[ERROR] D:\Arin\PlayGround\FirstClass.java:10:3: The field 'code' must be declared final. [MutableException]
[ERROR] D:\Arin\PlayGround\FirstClass.java:28:3: The field 'code' must be declared final. [MutableException]
Audit done.
Checkstyle ends with 2 errors.

Reply:
here, 

PS D:\Arin\PlayGround> type config.xml     
&lt;?xml version="1.0"?&gt;
&lt;!DOCTYPE module PUBLIC
  &emsp; "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
  &emsp; "https://checkstyle.org/dtds/configuration_1_3.dtd"&gt;

&lt;module name="Checker"&gt;
&emsp;&lt;module name="TreeWalker"&gt;
&emsp;&emsp;&lt;module name="MutableException"&gt;
&emsp;&emsp;&emsp;&lt;property name="format" value="^.*Exception$" /&gt;
&emsp;&emsp;&lt;/module&gt;
&emsp;&lt;/module&gt;
&lt;/module&gt;

PS D:\Arin\PlayGround> type FirstClass.java
class FirstClass extends Exception {
&emsp;private int code;

&emsp;public FirstClass() {
&emsp;&emsp;code = 1;
&emsp; }
}

class MyException extends Exception {
&emsp;private int code;

 &emsp;public MyException() {
  &emsp;&emsp;code = 2;
  &emsp;}
}

class MyThrowable extends Throwable {
  &emsp;final int code;
  &emsp;String message;

  &emsp;public MyThrowable(int code, String message) {
    &emsp;&emsp;this.code = code;
    &emsp;&emsp;this.message = message;
  &emsp;}
}

class BadException extends java.lang.Exception {
  &emsp;int code;

  &emsp;public BadException(int code) {
    &emsp;&emsp;this.code = code;
  &emsp;}
}

PS D:\Arin\PlayGround> java -jar checkstyle-10.5-SNAPSHOT-all.jar -c config.xml FirstClass.java
Starting audit...
[ERROR] D:\Arin\PlayGround\FirstClass.java:10:3: The field 'code' must be declared final. [MutableException]
[ERROR] D:\Arin\PlayGround\FirstClass.java:28:3: The field 'code' must be declared final. [MutableException]
Audit done.
Checkstyle ends with 2 errors.

Reply:
> here,
> 
> PS D:\Arin\PlayGround> type config.xml <?xml version="1.0"?> <!DOCTYPE module PUBLIC   "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"   "https://checkstyle.org/dtds/configuration_1_3.dtd">
> 
> <module name="Checker">  <module name="TreeWalker">   <module name="MutableException">    <property name="format" value="^.*Exception$" />   </module>  </module> </module>
> 
> PS D:\Arin\PlayGround> type FirstClass.java class FirstClass extends Exception {  private int code;
> 
>  public FirstClass() {   code = 1;   } }
> 
> class MyException extends Exception {  private int code;
> 
>  public MyException() {   code = 2;  } }
> 
> class MyThrowable extends Throwable {  final int code;  String message;
> 
>  public MyThrowable(int code, String message) {   this.code = code;   this.message = message;  } }
> 
> class BadException extends java.lang.Exception {  int code;
> 
>  public BadException(int code) {   this.code = code;  } }
> 
> PS D:\Arin\PlayGround> java -jar checkstyle-10.5-SNAPSHOT-all.jar -c config.xml FirstClass.java Starting audit... [ERROR] D:\Arin\PlayGround\FirstClass.java:10:3: The field 'code' must be declared final. [MutableException] [ERROR] D:\Arin\PlayGround\FirstClass.java:28:3: The field 'code' must be declared final. [MutableException] Audit done. Checkstyle ends with 2 errors.

@romani  ?

Reply:
@romani, did you check ?

Reply:
@arinmodi use back ticks to make code blocks in markdown, it will make examples much easier to read and discern from other text.

Reply:
@nrmancuso, updated the code blocks

Reply:
thanks a lot, I checked all examples, they works like this, they apply properties by AND, it is correct.